### PR TITLE
Prototype numeric input scrubber

### LINKED ISSUE #1053: Add pointer-based slider UI for numeric design panel inputs
## Feature Description

As an editor I want the most (all?) numeric inputs to have a slider according to figma design

[figma](https://www.figma.com/file/Cd9iacVknHK62RO4pRwb68/Stories-Editor?node-id=9810%3A1080)

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance Criteria

1. Opacity has a slider 
1. Other numeric inputs has a slider (list to be provided)

### PR DESCRIPTION
**Pull request by wassgha**
<p align="center">Draggable numeric input</p>

## Summary

Prototype requested by @samitron7 to demo a dragging gesture for changing numeric values. Differentiates between a click (to manually edit the numeric input by keyboard) and a drag.

## To-do
- [ ] Clean up code (use a mapping between pixels and numeric values)
- [ ] Add min/max bounds
- [ ] Handle mouse movement outside of input container
- [ ] Fix race condition between updating value and submitting the form
- [ ] Touchscreen compatibility
- [ ] Restore ability to select text while focused

Closes #1053

### CODE REVIEW

**Comment by samitron7** [comment]
@wassgha thanks so much for getting this in and it's 1AM in the morning! Anyway I can play around with this myself?

**Comment by swissspidy** [comment]
@samitron7 You can use [the QA environment](https://stories-qa-wordpress-amp.pantheonsite.io/) I sent an email about the other day.

Check https://github.com/google/web-stories-wp/blob/master/docs/testing-environments.md for access and https://github.com/google/web-stories-wp/blob/master/docs/testing-qa.md#web-stories-tester-plugin for how to test  this particular change.

Basically, you log in on https://stories-qa-wordpress-amp.pantheonsite.io/ with your username, then select `#1775 - Prototype...` from the `Using Web Stories` UI in the upper right corner and then click "Activate Selection". It might take a few moments to reflect the changes. After that, head to the editor.

File: assets/src/edit-story/components/form/numeric.js [hunk 1]
```diff
@@ -112,11 +116,86 @@ function Numeric({
...
+        submitCallback(e);
+      } else if (!focused && ref.current.contains(e.target)) {
+        handleFocus();
+      }
+    },
+    [focused, scrubbed, handleFocus, ref, submitCallback]
+  );
+
+  const scrubProps = {
+    onMouseMove: useCallback(
```
Comment (dvoytenko) [inline assets/src/edit-story/components/form/numeric.js:165]: Can we do `onPointer*` events instead?

File: assets/src/edit-story/components/form/numeric.js [hunk 2]
```diff
@@ -112,11 +116,86 @@ function Numeric({
...
+        }
+        if (!scrubbed) {
+          setScrubbed(true);
+        }
+        changeCallback(value + e.movementX, e);
+      },
+      [scrubbing, scrubbed, focused, changeCallback, value]
+    ),
+    onMouseUp: mouseUpCallback,
+    onMouseDown: (e) => {
```
Comment (dvoytenko) [inline assets/src/edit-story/components/form/numeric.js:178]: IMHO it'd be nicer to move these methods up with `mouseUpCallback`.

File: assets/src/edit-story/components/form/numeric.js [hunk 3]
```diff
@@ -112,11 +116,86 @@ function Numeric({
...
+      setScrubbing(true);
+      setScrubbed(false);
+      // Clear this scrubbing event as soon as mouse is released
+      // `setTimeout` is currently required to not break functionality.
+      e.target.ownerDocument.addEventListener(
```
Comment (dvoytenko) [inline assets/src/edit-story/components/form/numeric.js:183]: Can we use pointer capture for this instead?

**Review by dvoytenko — CHANGES_REQUESTED** [review]
I think this looks pretty good.

**Pull request closed without merging**